02_bubble/brvishnu92/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	sorted := make([]int, len(s))
+	copy(sorted, s)
+	swapped := true
+	z := 0
+	if len(s) > 0 {
+		z = sorted[0]
+	}
+	for swapped {
+		swapped = false
+		for i, x := range sorted {
+			if x < z && i != 0 {
```

Comment:
Don't compare loop index inside loop, but in your case start iterating from `i := 1`, this way you can also get rid of `if len(s) > 0 {` ...

```go
	for swapped {
		swapped = false
		for i := 1; i < len(sorted); i++ {
			if sorted[i] < sorted[i-1] {
				swapped = true
				sorted[i-1], sorted[i] = sorted[i], sorted[i-1]
			}
		}
	}
```

I see that wikipedia suggests exactly your approach, so I can't argue that this is incorrect. 
For completeness, IMO the most basic bubble sort algorithm looks like this:
```go
	for i := len(s); i > 0; i-- {
		for j := 1; j < i; j++ {
			if s[j-1] > s[j] {
				s[j-1], s[j] = s[j], s[j-1]
			}
		}
	}
```
see how the inner loop only iterates until `i`? this is because after the first iteration the largest element will have "bubbled up" to the right most spot in the array - there is no more need to compare anything against this biggest element again.

This can then be optimised to break if no swap happens at all (ideal for already sorted input):
```go
	for i := len(s); i > 0; i-- {
		swapped := false
		for j := 1; j < i; j++ {
			if s[j-1] > s[j] {
				s[j-1], s[j] = s[j], s[j-1]
				swapped = true
			}
		}
		if !swapped {
			// If no elements were swapped, the slice is sorted so we can stop here
			return
		}
	}
```
IMO this is not what you have implemented ;)

Either case, please remove the loop index comparison inside the loop.

Reply:
Done. I was fixated on not having a loop with the index like (for i:=1) . Hence all the workaround on it :-) Fixed now.